Sources/StreamChat/Utils/Atomic_Tests.swift

```diff
@@ -23,6 +24,33 @@ class Atomic_Tests: StressTestCase {
         XCTAssertEqual(stringAtomicValue, nil)
     }
     
+    func test_Atomic_CAS() {
+        boolAtomicValue = false
+        XCTAssertFalse(_boolAtomicValue.compareAndSwap(old: true, new: false))
+        XCTAssertFalse(_boolAtomicValue.compareAndSwap(old: true, new: true))
+        XCTAssertTrue(_boolAtomicValue.compareAndSwap(old: false, new: true))
+        XCTAssertFalse(_boolAtomicValue.compareAndSwap(old: false, new: true))
+    }
+    
+    func test_Atomic_CAS_Concurrent() {
+        boolAtomicValue = false
+        var swaps = 0
+
+        for _ in 0..<numberOfTestCycles {
+            DispatchQueue.random.async {
+                if self._boolAtomicValue.compareAndSwap(old: false, new: true) {
+                    DispatchQueue.main.async {
+                        swaps += 1
+                    }
+                }
+            }
+        }
+        
+        /// sleep for 1ms to avoid AssertAsync to quit too early (rare but possible)
+        usleep(1000)
+        AssertAsync.willBeEqual(swaps, 1)
+    }
```

Comment:
Since we don't need CAS for our specific problem, I would remove this, since we might not need it at all in the future